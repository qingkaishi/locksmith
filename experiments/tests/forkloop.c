#include <pthread.h>
#include <stdio.h>

void * dostuff(void * a) {
  *(int *)a = 0;
}

int main() {
  pthread_t t1;
  int i = 1; // not shared yet
  while(1) {
    pthread_create(&t1, NULL, dostuff, &i);
  }
}


