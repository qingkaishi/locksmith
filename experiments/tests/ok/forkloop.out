************************* STARTING *************************

shared: &i:forkloop.c:10

Warning: Possible data race: &i:forkloop.c:10 is not protected!
 references:
  dereference of cast:forkloop.c:5 at forkloop.c:5
    &i:forkloop.c:10 => cast:forkloop.c:5 at dostuff forkloop.c:12
  locks acquired:
    <empty>
  in: main at forkloop.c:8 


*************************** DONE ***************************
